### PR TITLE
Support for ConstFetch (true, false, null) in NodeFactory::createArrayItem

### LINKED ISSUE #7378: NodeFactory->createArray doesn't support boolean Node elements
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.13.2             |

## Minimal PHP Code Causing Issue
```php
<?php

class MyRule extends AbstractRector
{
    public function getNodeTypes(): array
    {
        return [Node\Expr\MethodCall::class];
    }

    public function refactor(Node $node)
    {
        // String works!
//        return $this->nodeFactory->createArray([new Node\Scalar\String_('hello')]);

        // Boolean isn't :(
        return $this->nodeFactory->createArray([$this->nodeFactory->createTrue()]);

        return null;
    }

    public function getRuleDefinition(): RuleDefinition
    {
        return new RuleDefinition('', []);
    }
}
```

## Expected Behaviour
It generates a node with array `[true]`

## Actual

Exception:
```
In NodeFactory.php line 517:
                                                                               
  Not implemented yet. Go to "Rector\Core\PhpParser\Node\NodeFactory::createArrayItem()" and add check for "PhpParser\Node\Expr\ConstFetch" node.     
```

### PR DESCRIPTION
**Pull request by xorik**
Fix for https://github.com/rectorphp/rector/issues/7378

---

EDIT by @TomasVotruba : Fixes https://github.com/rectorphp/rector/issues/7378 (to autoclose on merge)

### CODE REVIEW

File: tests/PhpParser/Node/NodeFactoryTest.php [hunk 1]
```diff
@@ -39,16 +41,47 @@ public function testCreateArray(array $inputArray, Array_ $expectedArrayNode): v
      */
     public function provideDataForArray(): Iterator
```
Comment (TomasVotruba) [inline tests/PhpParser/Node/NodeFactoryTest.php:42]: You can change the `@return` type to `Iterator<mixed>`, as the return data here are various :)

**Comment by TomasVotruba** [comment]
Thank you :clap: :+1: 

Amazing work :slightly_smiling_face:

**Comment by xorik** [comment]
No, thank you for the rector!
I'm migrating a huge app with 800+ services from self-made DI to a Symfony DI. Without rector it would take months...

**Comment by TomasVotruba** [comment]
Wow, that sounds huge :+1: 

> I'm migrating a huge app with 800+ services from self-made DI to a Symfony DI. 

I'd love to see that, feels like a dream job :)

**Pull request merged**